### ISSUE TITLE
Datastore index failes to create

### ISSUE THREAD

**Issue by hacktobeer** [forseti-security/osdfir-infrastructure#13]
"Error creating Index: googleapi: Error 403: The caller does not have permission"

* datastore api enabled in project
* service account terraform is running under has sufficient permission...I think (double checking now)

**Comment by hacktobeer** [comment]
Other people seem to have problems as well
https://github.com/hashicorp/terraform-provider-google/issues/7799

**Comment by hacktobeer** [comment]
This seems different, we need some more permission to be able to create indexes in datastore.

**Comment by hacktobeer** [comment]
Assigning datastore.owner or datastore.indexAdmin to the service account does not solve the error....weird.

**Comment by hacktobeer** [comment]
Probably a terraform datastore module bug....
If this does not get fixed soon I propose we roll-back to using modules/turbinia/data/index.yaml  for the datastore setup as changed in pull request #10

**Comment by hacktobeer** [comment]
FYI @aarontp

**Comment by aarontp** [comment]
I just re-tested a deployment, and it seemed to create the indexes just fine, so I think this may be resolved.  I'll close it for now, but if you see this happening again we can reopen and do more investigation.

**Comment by hacktobeer** [comment]
Re-opening as not working. Did you test this *without* the --use-gcloud-auth option for the deploy.sh script?

**Comment by hacktobeer** [comment]
Second related issue is that with the current datastore creation approach it failes if they already exist....

Can we consider rolling the datastore approach back to how it was (that worked...)?

**Comment by hacktobeer** [comment]
Re-opening as the fix (#20) does not work when using a service account (as opposed to using gcloud auth).
```
Error creating Index: googleapi: Error 403: The caller does not have permission

```

**Comment by hacktobeer** [comment]
Adding the below roles to the service account does not fix the error:
roles/datastore.owner
or
roles/datastore.indexAdmin

**Comment by hacktobeer** [comment]
Oh no, please say it's not like below.....
https://github.com/terraform-google-modules/terraform-google-cloud-datastore/blob/master/scripts/create-indexes.sh

**Comment by hacktobeer** [comment]
Fixed with #23
